File: front/vue3-vant-mobile/src/api/modules/schedule.ts
import request from '@/utils/request'
import type { ApiResponse } from '@/types/api'

export interface ScheduleEventItem {
  id: string
  userId: string
  title: string
  startAt: string
  endAt: string
  note?: string | null
  recurrence: 'none' | 'daily' | 'weekly'
  recurrenceEndDate?: string | null
  linkedEventId?: string | null
  instanceKey: string
  pendingInviteId?: string | null
  createdAt?: string
}

export function fetchMyScheduleEvents(params: { from: string, to: string }) {
  return request.get<ApiResponse<ScheduleEventItem[]>>('/schedule/events', { params })
}

export function fetchPartnerScheduleEvents(params: { from: string, to: string }) {
  return request.get<ApiResponse<ScheduleEventItem[]>>('/schedule/events/partner', { params })
}

export function createScheduleEvent(data: {
  title: string
  startAt: string
  endAt: string
  note?: string
  recurrence?: 'none' | 'daily' | 'weekly'
  recurrenceEndDate?: string
}) {
  return request.post<ApiResponse<ScheduleEventItem>>('/schedule/events', data)
}

export function updateScheduleEvent(id: string, data: {
  title?: string
  startAt?: string
  endAt?: string
  note?: string
  recurrence?: 'none' | 'daily' | 'weekly'
  recurrenceEndDate?: string
}) {
  return request.patch<ApiResponse<ScheduleEventItem>>(`/schedule/events/${id}`, data)
}

export function deleteScheduleEvent(id: string) {
  return request.delete<ApiResponse<void>>(`/schedule/events/${id}`)
}

export function inviteScheduleEvent(id: string) {
  return request.post<ApiResponse<void>>(`/schedule/events/${id}/invite`)
}

export function acknowledgeScheduleInvite(id: string, action: 'accept' | 'reject') {
  return request.post<ApiResponse<ScheduleEventItem>>(`/schedule/invites/${id}/acknowledge`, { action })
}

File: python/agent/src/life_assistant_agent/tools/schedule.py
"""HTTP helpers for schedule events."""

from typing import Any

from ..client import JavaClient


def _ensure_full_datetime(val: str | None) -> str | None:
    if val is None:
        return None
    if "T" not in val:
        return val + "T00:00:00"
    return val


async def schedule_list(
    client: JavaClient, from_: str, to: str, scope: str = "me",
) -> Any:
    params = {
        "from": _ensure_full_datetime(from_),
        "to": _ensure_full_datetime(to),
    }
    path = "/schedule/events/partner" if scope == "partner" else "/schedule/events"
    return await client.get(path, params)


async def schedule_create(
    client: JavaClient,
    title: str,
    start_at: str,
    end_at: str,
    note: str | None = None,
    recurrence: str | None = None,
    recurrence_end_date: str | None = None,
) -> Any:
    body: dict[str, Any] = {
        "title": title,
        "startAt": _ensure_full_datetime(start_at),
        "endAt": _ensure_full_datetime(end_at),
    }
    if note is not None:
        body["note"] = note
    if recurrence is not None:
        body["recurrence"] = recurrence
    if recurrence_end_date is not None:
        body["recurrenceEndDate"] = recurrence_end_date
    return await client.post("/schedule/events", body)


async def schedule_update(
    client: JavaClient,
    id: str,
    title: str | None = None,
    start_at: str | None = None,
    end_at: str | None = None,
    note: str | None = None,
    recurrence: str | None = None,
    recurrence_end_date: str | None = None,
) -> Any:
    body: dict[str, Any] = {}
    if title is not None:
        body["title"] = title
    v = _ensure_full_datetime(start_at)
    if v is not None:
        body["startAt"] = v
    v = _ensure_full_datetime(end_at)
    if v is not None:
        body["endAt"] = v
    if note is not None:
        body["note"] = note
    if recurrence is not None:
        body["recurrence"] = recurrence
    if recurrence_end_date is not None:
        body["recurrenceEndDate"] = recurrence_end_date
    return await client.patch(f"/schedule/events/{id}", body)
